src/System.Text.JsonLab.Dynamic/System/Text/Json/JsonDynamicObject.cs

```diff
@@ -218,6 +291,67 @@ private static unsafe bool TryEncodeControlChar(SymbolTable symbolTable, byte va
             return symbolTable.TryEncode(value, buffer, out written);
         }
 
+        private static int GetHashCode(ReadOnlySpan<byte> span)
+        {
+            int hash = 17;
+            foreach (byte element in span)
+            {
+                hash = hash * 31 + element;
+            }
+            return hash;
+        }
+
+        private static Dictionary<int, PropertyInfoLinkedList> GetTypeMap(Type type)
+        {
+            var dictionary = new Dictionary<int, PropertyInfoLinkedList>();
```

Comment:
Long term you might want to play with custom comparers that would work better for matching property names. Also, you might consider writing a custom hashtable optmized for this scenario.